packages/video_player/video_player_avfoundation/lib/src/avfoundation_video_player.dart

```diff
@@ -0,0 +1,172 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:video_player_platform_interface/video_player_platform_interface.dart';
+
+import 'messages.dart';
+
+/// An iOS implementation of [VideoPlayerPlatform] that uses method channels.
+class AVFoundationVideoPlayer extends VideoPlayerPlatform {
+  final VideoPlayerApi _api = VideoPlayerApi();
+
+  /// Registers this class as the default instance of [VideoPlayerPlatform].
+  static void registerWith() {
+    VideoPlayerPlatform.instance = AVFoundationVideoPlayer();
+  }
```

Comment:
This is new code so we should have a test for it.  Ideally we'd have an integration test.  This is particularly important since if this code was deleted today everything would technically still work, no?

Reply:
Thanks, I forgot to add the usual unit test for this.

> Ideally we'd have an integration test.

The integration tests implicitly exercise this code, because the actual registration logic runs. It's what's setting the implementation that `MiniController` (and thus all of these tests) are relying on.

> This is particularly important since if this code was deleted today everything would technically still work, no?

Today it would work, because everything is currently the same as the default method channel. As soon as we update Pigeon, if `registerWith` weren't there everything would fail because we'd be using the Dart side of the old API, and the native side of the new one.

But I've added an explicit integration test for now, just to make 100% sure we're not using the default implementation.

Reply:
> Today it would work, because everything is currently the same as the default method channel.

Which is what was happening, because I forgot to add the pubspec keys to register the Dart code 🤦🏻